cmd/fpga_plugin/README.md

```diff
@@ -1,120 +1,269 @@
+
+    The webhook can also set environment variables to instruct the FPGA prestart hook to program the FPGA
+    before launching the container.
+
 -   [FPGA prestart CRI-O hook](../fpga_crihook/README.md)
 
+    A [CRI-O](https://github.com/cri-o/cri-o) prestart hook that, upon instruction from the FPGA admission
+    controller, allocates and programs the FPGA before the container is launched.
 
-### Get source code:
-```
-    $ mkdir -p $GOPATH/src/github.com/intel/
-    $ cd $GOPATH/src/github.com/intel/
-    $ git clone https://github.com/intel/intel-device-plugins-for-kubernetes.git
-```
+The repository also contains an [FPGA helper tool](../fpga_tool/README.md) that may be useful during
+development, initial deployment and debugging.
 
-### Build FPGA device plugin:
-```
-    $ cd $GOPATH/src/github.com/intel/intel-device-plugins-for-kubernetes
-    $ make fpga_plugin
-```
+# Installation
+
+The below sections cover how to obtain, build and install this component.
+
+Components can generally be installed either using DaemonSets or running them
+'by hand' on each node.
+
+## Pre-built images
+
+Pre-built images of the components are available on the [Docker hub](https://hub.docker.com/u/intel).
+These images are automatically built and uploaded from the latest `master` branch of this repository.
+
+Release tagged images of the components are also available on the Docker hub, tagged with their
+release version numbers.
+
+The deployment YAML files supplied with these components in this repository use the images with the
+`devel` tag by default. If you do not build your own local images, then your Kubernetes cluster may
+pull down the `devel` images from the Docker hub by default.
+
+To use the release tagged versions of the images, edit the YAML deployment files appropriately.
+
+The following images are available on the Docker hub:
+
+- [The FPGA plugin](https://hub.docker.com/r/intel/intel-fpga-plugin)
+- [The FPGA admisson webhook](https://hub.docker.com/r/intel/intel-fpga-admissionwebhook)
+- [The FPGA CRI-O plugin (in the `initcontainer` image)](https://hub.docker.com/r/intel/intel-fpga-initcontainer)
+
+## Dependencies
+
+All components have the same basic dependancies as the
+[generic plugin framework dependencies](../../README.md#about)
+
+To obtain a fully operational FPGA enabled cluster, you must install all three
+major components:
+
+-   [FPGA device plugin](README.md) (this component)
+-   [FPGA admission controller webhook](../fpga_admissionwebhook/README.md)
+-   [FPGA prestart CRI-O hook](../fpga_crihook/README.md)
+
+The CRI-O hook is only *required* if orchestrated FPGA bitstream programming (mode) is
+being used, but is installed by default by the
+[FPGA plugin DaemonSet YAML](../../deployments/fpga_plugin/fpga_plugin.yaml). If not using
+the FPGA admission controller webhook orchestration mode, and therefore not using the
+CRI-O prestart hook, runtimes other than CRI-O can be used.
```

Comment:
The main point here is that default runtime (Docker) can be used in preprogrammed mode. This simplifies cluster setup a lot. Instead of fighting with CRI-O and its hooks user can use default kubernetes runtime.